staging/src/k8s.io/apimachinery/pkg/runtime/serializer/cbor/cbor.go

```diff
@@ -0,0 +1,304 @@
+	encMode, err := cbor.EncOptions{
+		// Map keys need to be sorted to have deterministic output, and this is the order
+		// defined in RFC 8949 4.2.1 "Core Deterministic Encoding Requirements".
+		Sort: cbor.SortBytewiseLexical,
+
+		// CBOR supports distinct types for IEEE-754 float16, float32, and float64. Store
+		// floats in the smallest width that preserves value so that equivalent float32 and
+		// float64 values encode to identical bytes, as they do in a JSON
+		// encoding. Satisfies one of the "Core Deterministic Encoding Requirements".
+		ShortestFloat: cbor.ShortestFloat16,
+		NaNConvert:    cbor.NaNConvert7e00,
+		InfConvert:    cbor.InfConvertFloat16,
+
+		// Prefer encoding math/big.Int to one of the 64-bit integer types if it fits.
+		BigIntConvert: cbor.BigIntConvertShortest,
+
+		// MarshalJSON for time.Time writes RFC3339 with nanos.
+		Time: cbor.TimeRFC3339Nano,
+
+		// Don't add tag 1 when encoding time.Time. For JSON parity, we can't require it
+		// when decoding to time.Time (in manifests and in dynamic client code it looks like
+		// any string), and decoding to interface{} must produce a string.
+		TimeTag: cbor.EncTagNone,
+
+		// Indefinite-length items have multiple encodings and aren't being used anyway, so
+		// disable to avoid an opportunity for nondeterminism.
+		IndefLength: cbor.IndefLengthForbidden,
+
+		// Preserve distinction between nil and empty for slices and maps.
+		NilContainers: cbor.NilContainerAsNull,
+	}.EncMode()
+	if err != nil {
+		panic(err)
+	}
+	return encMode
+}()
+
+var decMode cbor.DecMode = func() cbor.DecMode {
+	decMode, err := cbor.DecOptions{
+		// Maps with duplicate keys are well-formed but invalid according to the CBOR spec
+		// and never acceptable. Unlike the JSON serializer, inputs containing duplicate map
+		// keys are rejected outright and not surfaced as a strict decoding error.
+		DupMapKey: cbor.DupMapKeyEnforcedAPF,
+
+		// For JSON parity, decoding an RFC3339 string into time.Time needs to be accepted
+		// with or without tagging. If a tag number is present, it must be valid.
+		TimeTag: cbor.DecTagOptional,
+
+		// Observed depth up to 16 in fuzzed batch/v1 CronJobList. JSON implementation limit
+		// is 10000.
+		MaxNestedLevels: 64,
+
+		MaxArrayElements: 1024,
+		MaxMapPairs:      1024,
+
+		// Indefinite-length sequences aren't produced by this serializer, but other
+		// implementations can.
+		IndefLength: cbor.IndefLengthAllowed,
+
+		// Accept inputs that contain CBOR tags.
+		TagsMd: cbor.TagsAllowed,
+
+		// Decode type 0 (unsigned integer) as int64.
+		// TODO: IntDecConvertSigned errors on overflow, JSON will try to fall back to float64.
+		IntDec: cbor.IntDecConvertSigned,
+
+		// Disable producing map[cbor.ByteString]interface{}, which is not acceptable for
+		// decodes into interface{}.
+		MapKeyByteString: cbor.MapKeyByteStringForbidden,
+
+		// Error on map keys that don't map to a field in the destination struct.
```

Comment:
I thought JSON now warned, but did not fail.

Reply:
Yes, that's the default for field validation. There's an unimplemented test case in the alpha criteria that requires us to return strict decoding errors on unknown field. The library doesn't currently support reporting unknown fields without also aborting the decode, so until it does, I will have to capture this error and do a second pass with the lax decoding mode.

Reply:
Implemented support for strict decoding errors.